### ISSUE TITLE
Plugin is not truncating tables in between test calls (SQLite)

### ISSUE THREAD

**Issue by jamisonbryant** [vierge-noire/cakephp-test-suite-light#59]
Hi, I'm not sure if I'm doing something wrong or not but I'm noticing that the plugin is _not_ truncating tables after the test suite runs as indicated [in the setup document](https://github.com/vierge-noire/cakephp-fixture-factories/search?q=truncate). I am using SQLite as my test database driver, is this driver supported for this functionality?

Some example factory calls:
```php
// test 1: 5 entities persisted
AccessTokenFactory::make(['expires' => 0], 3)->withUserAndRole()->persist();
AccessTokenFactory::make(['expires' => 9999999999], 2)->withUserAndRole()->persist();

// test 2: 7 entities persisted
AccessTokenFactory::make(['expires' => 9999999999], 4)->withUserAndRole(['active' => 0])->persist();
AccessTokenFactory::make(['expires' => 9999999999], 3)->withUserAndRole()->persist();

// test 3: 6 entities persisted
AccessTokenFactory::make(['expires' => 9999999999], 2)->withUserAndRole(['login_allowed_after' => FrozenTime::now()->addDay()])->persist();
AccessTokenFactory::make(['expires' => 9999999999], 4)->withUserAndRole()->persist();

// test 4: 4 entities persisted
AccessTokenFactory::make(['expires' => 9999999999], 3)->withUserAndRole()->persist();
AccessTokenFactory::make(['expires' => 9999999999], 1)->withUserAndRole(roleParameters: ['slug' => 'disallowed_slug'])->persist();
```

Total entities: 22. SQLite shows 26 rows in my `access_tokens` table, so I presume some were created as side-effects of other tests running. But, the table is fully populated at the end of the test suite, leading me to believe it is NOT truncated between tests. I have to add this line in my teardown function to get my test suite to pass:

```php
protected function tearDown(): void
{
        // Manual truncation
        $this->AccessTokens->deleteAll([1 => 1]);

        unset($this->AccessTokens);
        unset($this->validator);
        parent::tearDown();
}
```

I noticed this is happening for my other factories as well, `ClientFactory` (2 rows) and `UserFactory` (12 rows).

I'm new to the plugin so I'm unsure how to further debug it - is there any additional information I need to provide? I would expect these tables to be completely empty at the end of the test run (without my manual truncation).

**Comment by jamisonbryant** [comment]
I was initially confused when this issue jumped repos, but thanks for putting it in the right place :) 

> I assume you are using a CakePHP version below 4.3, since you are quoting that part of the docs.

- I am actually using CakePHP v4.4.5 on PHP 8.1.9. 
- My composer.json shows the constraint `^2.5` for `vierge-noire/cakephp-fixture-factories` and it appears the actual installed version is 2.7.1. 
- That package shows a dependency of `vierge-noire/cakephp-test-suite-light ^2.1` in my tree as well with an actual installed version of 2.4.

> The test suite light truncates tables before each tests and not after. Therefore test DB will not be emptied at the end of a test, but rather at the beginning of the next one.

It does, and I was trying to come up with a means of replicating my issue but I'm having difficulty. What I'm observing is as follows, maybe you can spot the problem:

- Running each test one at a time (manually) DOES truncate the table in between tests, as expected. All tests pass in this case.
- Running ALL of the tests in the whole class (manually) DOES NOT truncate the table, I know this because my table ends up with ~26 rows, but the final test in the class should create only 4 rows. Several tests fail when run this way.
- Running the entire test suite results in ~28 rows in the table, and some tests fail (the same ones as in the previous bullet)

I am running the tests with both PhpStorm's built-in buttons for running per-function and per-class tests, but also plain old phpunit CLI. I suppose to summarize what I'm observing I would say "the tests pass in isolation but when run all together some fail because the database is not being cleaned up in between test runs".

Hope this is more helpful.

**Comment by pabloelcolombiano** [comment]
Thanks for the complete description.

Is your class using the TruncateDirtyTables trait as described here?
https://github.com/vierge-noire/cakephp-test-suite-light#with-cakephp-43

**Comment by jamisonbryant** [comment]
Yikes, it wasn't. Adding the trait clears up the problem. I didn't even know I had to take that step.

I came to this repo by the fixture factories repo, which actually doesn't mention anything about a trait (I checked, I grepped). Can [setup.md](https://github.com/vierge-noire/cakephp-fixture-factories/blob/main/docs/setup.md) be updated to mention that this trait is required for CakePHP 4.3+? The instructions are great for all other versions.
